### PR TITLE
Initial docs commit with contributing guidelines

### PR DESCRIPTION
**Pull request by dcmiddle**
Added initial contributing guidelines and community links.
I've optimistically assumed a mail list name. I also optimistically refer to Jira. (Both presume incubated project status rather than a lab.)
For chat and source code though I've provided the existing links (i.e. crypto-lib).
~I'll add in a sphinx Makefile as a separate PR.~
Added build script as another commit to this PR

### CODE REVIEW

**Review by mikelodder7 — APPROVED** [review]

**Review by hartm — APPROVED** [review]
Looks fantastic.  Thanks!

File: README.md [hunk 1]
```diff
@@ -85,15 +85,25 @@ System OpenSSL library is required.
   - set INDY_CRYPTO_PREBUILT_DEPS_DIR=C:\BIN\x64
   - set OPENSSL_DIR=C:\BIN\x64
 
-## API Documentation
+## Documentation
+Project documentation is available via sphinx.
+Install sphinx:
+```
+pip install -U Sphinx
```
Comment (mikelodder7) [inline README.md:92]: Why do we need sphinx?
Is this your preferred documentation generator?

**Review by mikelodder7 — COMMENTED** [review]
Do we want to be using Makefiles?

I thought we were going to try to use the Rust build system for this. At least that was my understanding after talking with @dhuseby

File: README.md (hunk 1 above)
Comment (dcmiddle) [reply README.md:92]: Yep. It's what we use for https://sawtooth.hyperledger.org/docs/

Comment (dcmiddle) [reply README.md:92]: And sphinx generates the Makefile. I wouldn't see us using Makefiles broadly in this project.

Comment (mikelodder7) [reply README.md:92]: okay as long as we are not committing to Makefiles for now I'm okay with it.

**Review by mikelodder7 — APPROVED** [review]

**Comment by dcmiddle** [comment]
Rebased over master.

**Pull request merged**